src/App.tsx

```diff
@@ -1,14 +1,67 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { getTodos } from './api';
+
+type Props = {
+  query: string;
+  status: TodoStatus;
+};
+
+const getFilteredTodos = (todos: Todo[], { query, status }: Props) => {
+  let filterTodos = todos;
+
+  if (query.trim()) {
+    filterTodos = filterTodos.filter(todo =>
+      todo.title.toLowerCase().includes(query.trim().toLowerCase()),
+    );
+  }
+
+  if (status !== TodoStatus.all) {
+    filterTodos = filterTodos.filter(todo =>
+      status === TodoStatus.completed ? todo.completed : !todo.completed,
+    );
+  }
+
+  return filterTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState(TodoStatus.all);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        return (
+          <p style={{ color: 'red' }}>Something went wrong, try again later</p>
+        );
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  const filterTodos = getFilteredTodos(todos, { query, status });
+
+  // const handleSelect = (todo: Todo | null) => {
+  //   setSelectedTodo(todo);
+  // };
+
+  // const handleClose = () => {
```

Comment:
it's better not to leave comments - just delete code you don't need anymore

BTW - handleClose function makes sense (it doesn't accept any arguments so you can just pass it as prop), when handleSelect just duplicated the original function

FYI only